### PR TITLE
chore: update github actions

### PR DESCRIPTION
**Pull request by ForsakenHarmony**
### Description

The `node16` runner is deprecated, I updated all the actions I could here.

Still missing:
~~https://github.com/pnpm/action-setup~~ (removed in favor of corepack, but that needs to be tested)
https://github.com/technote-space/get-diff-action (archived, needs to be replaced)
https://github.com/dependabot/fetch-metadata (has PR: https://github.com/dependabot/fetch-metadata/pull/443)
~~https://github.com/ts-graphviz/setup-graphviz (has PR: https://github.com/ts-graphviz/setup-graphviz/pull/558)~~
https://github.com/thomaseizinger/create-pull-request

### CODE REVIEW

**Review by mehulkar — COMMENTED** [review]
Kicking off some dry runs for CI workflows to validate:

- ❌ [Dry run for turborepo release](https://github.com/vercel/turbo/actions/runs/7834625823)
- [Benchmark](https://github.com/vercel/turbo/actions/runs/7834636474)
- ❌ [Library release](https://github.com/vercel/turbo/actions/runs/7834642627)

**Review by mehulkar — CHANGES_REQUESTED** [review]
Looks like there's some glibc missing issue for both our release workflows. We'd need to work those out before merge

**Review by mehulkar — APPROVED** [review]

File: .github/workflows/lsp.yml
[the remark this answers is not in the document]
Comment (arlyon) [reply .github/workflows/lsp.yml:1]: Answered in DM but repeating here; the LSP release is expected to fail in one case so I will resolve this.

File: .github/actions/setup-rust/action.yml [hunk 1]
```diff
@@ -58,16 +58,16 @@ runs:
     - name: Set Up Protoc
       id: set-up-protoc
       continue-on-error: true
-      uses: arduino/setup-protoc@v1.2.0
+      uses: arduino/setup-protoc@v3.0.0
       with:
-        version: "3.x"
+        version: "23.x"
         repo-token: ${{ inputs.github-token }}
```
Comment (kdy1) [inline .github/actions/setup-rust/action.yml:64]: I think we may add some comments describing that this is to avoid rate limit.

Comment (kdy1) [reply .github/actions/setup-rust/action.yml:64]: Mainly because passing token to `setup-rust` action looks weird.

**Review by kdy1 — APPROVED** [review]

**Comment by anthonyshew** [comment]
Will close this as stale.

**Pull request closed without merging**